x-pack/plugins/cloud_security_posture/server/saved_objects/migrations/csp_rule_template.ts

```diff
@@ -36,6 +37,20 @@ function migrateCspRuleMetadata(
   };
 }
 
+function deprecateCspRuleTemplateState(
+  doc: SavedObjectUnsanitizedDoc<CspRuleTemplateV840>,
+  context: SavedObjectMigrationContext
+): SavedObjectUnsanitizedDoc<CspRuleTemplateV870> {
+  const { ...metadata } = doc.attributes.metadata;
+  return {
+    ...doc,
+    attributes: {
+      metadata,
+    },
+  };
```

Comment:
This is an interesting approach, it is pretty straightforward.

Anyway, I gave it a try:
![image](https://user-images.githubusercontent.com/19221736/210359640-972f9b4d-b843-4ac5-90cf-593869fbb1a0.png)

This makes sense since what you suggest is actually cloning the same object, so the type remains.
And the fields remain read-only

Reply:
we don't need to delete from the original doc anyway, just return a new doc with the new interface